app/2.0/docs/tools/polymer-json.md

```diff
@@ -0,0 +1,171 @@
+
+In a Polymer app, the files listed in the `fragments` property usually contain one or more element definitions that may or may not be required during the user’s interaction with the app, and can thus be lazily loaded.
+
+### extraDependencies
+Optional
+Type: `Array` of `String`
+
+Dependencies that the analyzer component of the Polymer build toolchain can’t discover, and that do not need to be bundled.
+
+### shell
+*Required* 
+Type: `String`
+
+The app shell. Contains the main code that boots the app and loads necessary resources. The shell usually includes the common UI for the app, and the router, which loads resources needed for the current route.
+
+### sourceGlobs
+Optional, Defaults to `[“src/**/*”]`
+Type: `Array` of `String`
+
+An optional array of globs matching your application source files. If left out, defaults to all files in your project `src/` directory. You’ll need to set this if you store your source files in other locations.
+
+In the Shop app, source files are stored in `/src`, `/data` and `/images`:
+
+```
+-shop
+  |-README.md
+  |-app.yaml
+  |-bower.json
+  |-build
+  |-data
+  |-images
+  |-index.html
+  |-manifest.json
+  |-polymer.json
+  |-service-worker.js
+  |-src
+  |-sw-precache-config.js
+  |-test
+```
+
+The `sourceGlobs` property is set as follows:
+
+```json
+"sourceGlobs": [
+   "src/**/*",
+   "data/**/*",
+   "images/**/*",
+   "bower.json"
+  ],
+```
+
+### builds
+Optional
+Type: `Array` of `Build Configuration` objects
+
+You can configure how the CLI builds your application via the `builds` property. This is equivalent to passing different CLI flags to the build command, but storing them here will configure the build for every run:
+
+* `name`: An optional name for your build. If multiple builds are defined, the `name` property is required.
+* `addServiceWorker`: If `true`, generate a service worker for your application.
+* `swPrecacheConfig`: An optional configuration file for the generated service worker.
+* `insertPrefetchLinks`: If `true`, insert prefetch link elements into your fragments so that all dependencies are prefetched immediately.
+* `bundle`: If `true`, bundle your application.
+* `html`:
+** `minify`: If `true`, minify all HTML.
+* `css`:
+** `minify`: If `true`, minify all CSS.
+* `js`:
+** `minify`: If `true`, minify all JS.
+** `compile`: If `true`, use babel to compile all ES6 JS down to ES5.
+
+As an example, here is the configuration for a bundled, minified application build:
+
+```json
+"builds": [{
+    "bundle": true,
+    "js": {"minify": true},
+    "css": {"minify": true},
+    "html": {"minify": true}
+  }]
+```
+
+And here is a configuration to generate two optimized builds: One bundled and one unbundled:
+
+```json
+"builds": [{
+    “name”: “bundled”,
+    "bundle": true,
+    "js": {"minify": true},
+    "css": {"minify": true},
+    "html": {"minify": true}
+  },{
+    “name”: “unbundled”,
```

Comment:
bad quotation mark symbols got copied into here, will break other users who copy-paste this